prow/plugins/lgtm/lgtm.go

```diff
@@ -208,55 +205,14 @@ func handle(wantLGTM bool, config *plugins.Configuration, ownersClient repoowner
 		return gc.CreateComment(rc.repo.Owner.Login, rc.repo.Name, rc.number, plugins.FormatResponseRaw(rc.body, rc.htmlURL, rc.author, resp))
 	}
 
-	// Determine if reviewer is already assigned
-	isAssignee := false
-	for _, assignee := range assignees {
-		if assignee.Login == author {
-			isAssignee = true
-			break
-		}
-	}
-
 	// check if skip collaborators is enabled for this org/repo
 	skipCollaborators := skipCollaborators(config, org, repoName)
 
-	// either ensure that the commentor is a collaborator or an approver/reviwer
-	if !isAuthor && !isAssignee && !skipCollaborators {
-		// in this case we need to ensure the commentor is assignable to the PR
-		// by assigning them
-		log.Infof("Assigning %s/%s#%d to %s", org, repoName, number, author)
-		if err := gc.AssignIssue(org, repoName, number, []string{author}); err != nil {
-			msg := "assigning you to the PR failed"
-			if ok, merr := gc.IsCollaborator(org, repoName, author); merr == nil && !ok {
-				msg = fmt.Sprintf("only %s/%s repo collaborators may be assigned issues", org, repoName)
-			} else if merr != nil {
-				log.WithError(merr).Errorf("Failed IsCollaborator(%s, %s, %s)", org, repoName, author)
-			} else {
-				log.WithError(err).Errorf("Failed AssignIssue(%s, %s, %d, %s)", org, repoName, number, author)
-			}
-			resp := "changing LGTM is restricted to assignees, and " + msg + "."
-			log.Infof("Reply to assign via /lgtm request with comment: \"%s\"", resp)
-			return gc.CreateComment(org, repoName, number, plugins.FormatResponseRaw(body, htmlURL, author, resp))
-		}
-	} else if !isAuthor && skipCollaborators {
-		// in this case we depend on OWNERS files instead to check if the author
-		// is an approver or reviwer of the changed files
-		log.Debugf("Skipping collaborator checks and loading OWNERS for %s/%s#%d", org, repoName, number)
-		ro, err := loadRepoOwners(gc, ownersClient, org, repoName, number)
-		if err != nil {
-			return err
-		}
-		filenames, err := getChangedFiles(gc, org, repoName, number)
-		if err != nil {
-			return err
-		}
-		if !loadReviewers(ro, filenames).Has(github.NormLogin(author)) {
-			resp := "adding LGTM is restricted to approvers and reviewers in OWNERS files."
-			log.Infof("Reply to /lgtm request with comment: \"%s\"", resp)
-			return gc.CreateComment(org, repoName, number, plugins.FormatResponseRaw(body, htmlURL, author, resp))
-		}
+	// check if the commentor is allowed to change LGTM
+	ok, err := checkAllowedToLGTM(wantLGTM, rc, skipCollaborators, gc, ownersClient, log)
+	if !ok || err != nil {
```

Comment:
We are ignoring the error `if ok`? I think this is currently ok since we never comment if it's ok, and the error is for the comment? Maybe this could be factored differently